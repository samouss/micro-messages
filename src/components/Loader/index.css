/**
 * @NOTE: Loader from http://tobiasahlin.com/spinkit/
 */

.Loader {
  width: 20px;
  height: 20px;
  margin: 20px auto;
  position: relative;
  align-self: center;
}

.Loader__bounce-1,
.Loader__bounce-2 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: #bcbcbc;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  animation: sk-bounce 2.0s infinite ease-in-out;
}

.Loader__bounce-2 {
 -webkit-animation-delay: -1.0s;
 animation-delay: -1.0s;
}

@-webkit-keyframes sk-bounce {
 0%, 100% { -webkit-transform: scale(0.0) }
 50% { -webkit-transform: scale(1.0) }
}

@keyframes sk-bounce {
 0%, 100% {
   transform: scale(0.0);
   -webkit-transform: scale(0.0);
 } 50% {
   transform: scale(1.0);
   -webkit-transform: scale(1.0);
 }
}
