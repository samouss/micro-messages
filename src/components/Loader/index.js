// @flow

import React from 'react';
import './index.css';

const Loader = () => (
  <div styleName="Loader">
    <div styleName="Loader__bounce-1" />
    <div styleName="Loader__bounce-2" />
  </div>
);

export default Loader;
