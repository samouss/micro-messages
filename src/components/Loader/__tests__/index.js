// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../index';

describe('<Loader />', () => {
  it('expect to render the component correctly', () => {
    const component = shallow(
      <Loader />,
    );

    expect(component).toMatchSnapshot();
  });
});
