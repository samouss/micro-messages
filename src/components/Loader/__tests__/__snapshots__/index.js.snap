// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> expect to render the component correctly 1`] = `
<div
  styleName="Loader"
>
  <div
    styleName="Loader__bounce-1"
  />
  <div
    styleName="Loader__bounce-2"
  />
</div>
`;
